app/main.py

```
@@ -1,3 +1,64 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: object) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(
+                self.km + other.km
+            )
+        else:
+            return Distance(
+                self.km + float(other)
+            )
+
+    def __iadd__(self, other: object) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += float(other)
+        return self
+
+    def __mul__(self, other: object) -> Distance:
+        return Distance(self.km * other)
```

This is an error: you should check if `other` is a number (int or float) before multiplying. Also, multiplying by another `Distance` object is not defined and should probably raise an error.